src/Application/Features/TodoItems/Queries/GetAllTodoItems/GetAllTodoItemsQuery.cs

```diff
@@ -9,22 +9,26 @@ public record GetAllTodoItemsQuery : IRequest<IReadOnlyList<TodoItemDto>>;
 public class GetAllTodoItemsQueryHandler : IRequestHandler<GetAllTodoItemsQuery, IReadOnlyList<TodoItemDto>>
 {
     private readonly IMapper _mapper;
-    private readonly IReadRepositoryBase<TodoItem> _repository;
+    private readonly IApplicationDbContext _dbContext;
 
     public GetAllTodoItemsQueryHandler(
         IMapper mapper,
-        IReadRepositoryBase<TodoItem> repository)
+        IApplicationDbContext dbContext)
     {
         _mapper = mapper;
-        _repository = repository;
+        _dbContext = dbContext;
     }
 
     public async Task<IReadOnlyList<TodoItemDto>> Handle(
         GetAllTodoItemsQuery request,
         CancellationToken cancellationToken)
     {
         var spec = new AllTodoItemSpec();
-        var items = await _repository.ListAsync(spec, cancellationToken);
+
+        var items = await _dbContext.TodoItems
+            .WithSpecification(spec)
+            .ToListAsync(cancellationToken);

```

Comment:
While we are still using automapper we need to use `ProjectTo()` to apply the projection before we execute the query.